src/app/legacy/containers/CpsMetadata/index.test.jsx

```diff
@@ -1,12 +1,14 @@
 import React from 'react';
-import { render, waitFor } from '@testing-library/react';
+import { waitFor } from '@testing-library/react';
 import { RequestContextProvider } from '#contexts/RequestContext';
 import { ToggleContextProvider } from '#contexts/ToggleContext';
 import { articleDataNews } from '#pages/ArticlePage/fixtureData';
 import { ARTICLE_PAGE } from '#app/routes/utils/pageTypes';
 import { shouldMatchSnapshot } from '#psammead/psammead-test-helpers/src';
 import { ServiceContextProvider } from '../../../contexts/ServiceContext';
+import { render } from '../../../components/react-testing-library-with-providers';
```

Comment:
I think if you use `render` from `react-testing-library-with-providers` then you don't need to import and use `ThermeProvider`, it should be provided via the `render`

Reply:
Heya, so I have a question on this! In this example, if I just use `render` from `react-testing-library-with-providers` and don't use the the `ThemeProvider`, then the` shouldMatchSnapshot` tests in the suite fail.

Should I be refactoring these snapshot tests? I did this for the Error Page in
[src/app/pages/ErrorPage/index.test.jsx](https://github.com/bbc/simorgh/pull/10660/files#diff-f9876880984ef2c2b886ab3ff4019a4ce5f3ec4782364c60164505957ce9e028) so that I could add the `render` function.

Reply:
Yea I think thats on the right track using the `toMatchSnapshot` function instead of the Psammead `shouldMatchSnapshot` helper.

In your ErrorPage tests, if you import and use the `render` from `react-testing-library-with-providers`, you should be able to write the tests like:
```
  it('should correctly render for 404', () => {
    const { container } = render(<ErrorPage errorCode={404} />, {
      service: 'news',
    });

    expect(container).toMatchSnapshot();
  });
```

Instead of creating the mock `ErrorPageXXX` components for each error state. The `render` should be wrapping all the contexts, so you can pass in the options object as the second argument of `render` and pass in things like `service: 'news'`

Happy to go through in more detail on a call if needed!

Reply:
The `render` from `react-testing-library-with-providers` doesn't play well with the Psammead `shouldMatchSnapshot`, so using the built-in `toMatchSnapshot` is the best option. Hopefully shouldn't require major refactors/rewrites. Just rendering the component first, like the other tests, then writing an assert on it like `expect(container).toMatchSnapshot()`

Reply:
Thanks! Yes this makes sense, and that example helps a lot. I'll update these now.